### PR TITLE
Incorporate `law` and `mldatafind`

### PR DESCRIPTION
**Pull request by EthanMarx**
Incorporates recent `mldatafind` overhaul so we can now generate training and testing strain data via `law`. 

Updates the README with minimal instructions on getting started.

### CODE REVIEW

**Comment by deepchatterjeeligo** [comment]
can you separate out the sorting of the imports in a separate PR @EthanMarx

**Comment by EthanMarx** [comment]
yeah sure

**Comment by EthanMarx** [comment]
@deepchatterjeeligo Should be good for review now that pre commit is fixed

File: amplfi/law/pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+version = "0.1.0"
+description = "Law tasks for running amplfi analyses"
+authors = ["Ethan Jacob Marx <ethan.marx@ligo.org>"]
+packages = [{include = "amplfi"}]
+
+
+[tool.poetry.dependencies]
+python = ">=3.9,<3.12"
+law = {git = "https://github.com/riga/law.git", branch = "master"}
+mldatafind = {path = "../../mldatafind/", develop = true}
```
Comment (deepchatterjeeligo) [inline amplfi/law/pyproject.toml:12]: Do we have this submodule?

File: amplfi/law/amplfi/law/base.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import os
+
+from amplfi.law.paths import paths
+
+
+class AmplfiDataTaskMixin:
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
```
Comment (deepchatterjeeligo) [inline amplfi/law/amplfi/law/base.py:8]: What is the super class here?

File: amplfi/law/amplfi/law/base.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+        self.image = paths().container_root / "data.sif"
+
+    def sandbox_env(self, env):
+        # get `mldatafind.law.DataTask` environment variables
```
Comment (deepchatterjeeligo) [inline amplfi/law/amplfi/law/base.py:12]: Can you provide a use case example here i.e. what `env`

File: amplfi/law/pyproject.toml (hunk 1 above)
Comment (deepchatterjeeligo) [reply amplfi/law/pyproject.toml:12]: I see its added

**Review by deepchatterjeeligo — CHANGES_REQUESTED** [review]
I think this mostly looks fine. If you can remove the stray notebook, and explain some of the questions I have.

File: amplfi/law/amplfi/law/base.py (hunk 3 above)
Comment (EthanMarx) [reply amplfi/law/amplfi/law/base.py:12]: Yeah this is a `law` hook for mapping local environment variables into the container that the tasks are run inside. So for `mldatafind` these are things like credentials, segment database urls, etc.

File: amplfi/law/pyproject.toml (hunk 1 above)
Comment (EthanMarx) [reply amplfi/law/pyproject.toml:12]: yup

File: amplfi/law/amplfi/law/base.py (hunk 2 above)
Comment (EthanMarx) [reply amplfi/law/amplfi/law/base.py:8]: This is meant to be a "Mixin" for `law` `SandboxTasks`, specifically for the use case of appending environment variables. You can see it's use e.g. [here](https://github.com/ML4GW/amplfi/blob/19cdf5ba5646f7724f2ba0976ea1596e90f4a318/amplfi/law/amplfi/law/tasks.py#L24)

For reasons that are hard to explain over github, it didn't make sense to subclass the `mldatafind` tasks and override the `sandbox_env` due to additional boiler plate that would induce

File: amplfi/law/amplfi/law/Untitled-1.ipynb
[the remark this answers is not in the document]
Comment (EthanMarx) [reply amplfi/law/amplfi/law/Untitled-1.ipynb:1]: done

File: amplfi/law/amplfi/law/base.py (hunk 2 above)
Comment (deepchatterjeeligo) [reply amplfi/law/amplfi/law/base.py:8]: OK. But then add comments to which method in the law/mldatafind source code is being invoked when you call `super()`.

File: amplfi/law/amplfi/law/base.py (hunk 3 above)
Comment (deepchatterjeeligo) [reply amplfi/law/amplfi/law/base.py:12]: Let's add that as a docstring, otherwise its unclear.

Comment (EthanMarx) [reply amplfi/law/amplfi/law/base.py]: done

File: amplfi/law/amplfi/law/base.py (hunk 2 above)
Comment (EthanMarx) [reply amplfi/law/amplfi/law/base.py:17]: done

**Comment by EthanMarx** [comment]
@deepchatterjeeligo Just added some more documenation. Let me know if things are still unclear

**Review by deepchatterjeeligo — APPROVED** [review]

**Pull request merged**